### ISSUE TITLE
Question about general use in any case

### ISSUE THREAD

**Issue by Petros626** [LearnTechWithUs/Stereo-Vision#15]
Hello,

I would like to know, if this script can measure the distance of any object shown in a stream, no matter where it's located? 
If not, what adjustments in the code are necessary?

I plan to use a stereo camera and a pretrained CNN, which detects objects, additionally I want to measure the distance to the detected objects.

Thanks in advance

**Comment by shanearthur** [comment]
Hey Petros626,

The script will produce a two dimensional array of values which represent the depth of each pixel in the scene, however due to the nature of this traditional method, not all pixels will have values attributed to them. Some ill-posed regions (areas where the algorithm has a hard time determining depth) will be left empty.

![basicblockmatching1](https://github.com/LearnTechWithUs/Stereo-Vision/assets/59723062/9ed407ae-dbf4-4337-952a-335879baa42c)

As long as the object is within the image/frame from your stream and a value which represents the pixels of your object is within the depth map, you will be able to determine the estimated depth. I'd recommend doing some averaging of small kernels depending on what kind of object it is.

_Note:_ the resulting values will be **disparity** values, which you will use to calculate metric depth with `depth = baseline * focal_length / disparity`. For more see [here](https://docs.opencv.org/4.1.1/dd/d53/tutorial_py_depthmap.html).

**Comment by Petros626** [comment]
Thank you for your detailed answer. The most algorithms I saw was with a known object size, but this wasn't which I was searching for. So if with the mentioned formula you can calculate the distance why in your code specific values are made here:

https://github.com/LearnTechWithUs/Stereo-Vision/blob/597d9e5d4fdeb96583339365f41bf9488ea5c9fd/Main_Stereo_Vision_Prog.py#L37

**Comment by shanearthur** [comment]
> why in [the] code specific values are made here

Those are likely the specific parameters provided by datasets which the author was using to test this code. They may also be customized to compensate for the averaging being done two lines above your referenced line, where the author is getting the average disparity value of a small kernel of pixels around the pixel in question:
https://github.com/LearnTechWithUs/Stereo-Vision/blob/597d9e5d4fdeb96583339365f41bf9488ea5c9fd/Main_Stereo_Vision_Prog.py#L35

**Comment by Vujas-Eteph** [comment]
Yes, as @shanearthur said. Those values (polynomial coefficients) are estimated based on a custom dataset.
(Similar to issue https://github.com/LearnTechWithUs/Stereo-Vision/issues/5). 
When and how we estimated those parameters is shown in this section of the [YouTube video](https://youtu.be/xjx4mbZXaNc?t=15). We plotted a curve in which we had the distance on the *x-axis* and the disparity values on the *y_axis* (if I remember correctly). Afterward, we did a polynomial regression of degree 3 (known from the literature) via Excel - but you can do that with Python libraries to optimize the value.
